interview-questions.md

```diff
@@ -1,54 +1,26 @@
 # ASSESSMENT 4: Interview Practice Questions
 Answer the following questions.
-
 First, without external resources. Challenge yourself to answer from memory.
-
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.  
-
 1. What is an instance variable in Ruby? How is it different from other variables?
-
-  Your answer:
-
-  Researched answer:
-
-
-
+  Your answer: it has a name starting with @ and its scope is specific to whatever object its referred to.
+  Researched answer: instance variables do not have to be declared and its is dynamically connected to an object when it is first referenced.
 2. What is a block in Ruby?
-
-  Your answer:
-
-  Researched answer:
-
-
-
+  Your answer: It is a piece of code that accepts arguments and returns a value
+  Researched answer: it is like a method but it doesnt have a name and does not belong to an object. they can also be created by passing them through a method and then calling that method.
 3. Ruby has an implicit return. What does that mean?
-
-  Your answer:
-
+  Your answer: it will automatically return the value of the last line in the method
   Researched answer:
-
-
-
 4. What is object-oriented programming? How is it different than functional programming?
-
-  Your answer:
-
-  Researched answer:
-
-
-
+  Your answer: it is based on objects and can have data stored inside of them where as functional programming can not have data stored inside objects.
+  Researched answer: Importance is given to the data rather than procedures or functions because it works as a real world. POP follows the Top-Down approach.
 5. What is the difference between a class and an object?
-
-  Your answer:
-
-  Researched answer:
-
-
-
+  Your answer: an object is an instance of a class and classes are the blueprint where objects are created.
+  Researched answer: objects can hold any type of data
```

Comment:
I want to encourage you to expand on your researched answers. Objects can hold any time of data is cool, but what about the additional research regarding the difference between classes and objects?